frontend/gateway/gateway.go

```diff
@@ -137,31 +141,51 @@ func (gf *gatewayFrontend) Solve(ctx context.Context, llbBridge frontend.Fronten
 			}
 		}
 	} else {
-		sourceRef, err := reference.ParseNormalizedNamed(source)
+		c, err := forwarder.LLBBridgeToGatewayClient(ctx, llbBridge, opts, inputs, gf.workers, sid, sm)
 		if err != nil {
 			return nil, err
 		}
-
-		dgst, config, err := llbBridge.ResolveImageConfig(ctx, reference.TagNameOnly(sourceRef).String(), llb.ResolveImageConfigOpt{})
+		dc, err := dockerui.NewClient(c)
 		if err != nil {
 			return nil, err
 		}
-		mfstDigest = dgst
-
-		if err := json.Unmarshal(config, &img); err != nil {
+		st, dockerImage, err := dc.NamedContext(ctx, source, dockerui.ContextOpt{
+			CaptureDigest: mfstDigest,
```

Comment:
Might be missing something, but isn't `mfstDigest == nil` at this point, so it won't be captured?